### ISSUE TITLE
Pulsaudio creates second device after disconnection of BT-Box

### ISSUE THREAD

**Issue by LordMyschkin** [owntone/owntone-server#1265]
I am using the last version of forked-daapd from github; under elementary-OS (Ubuntu Clone with pulseaudio), when I turn of the Bluetooth speaker, and then turn it on again, the Speaker gets listed twice, once enabled (although not playing) and the second time disabled; I habe to enable it via mpc enable 

eg: 
Playing music
mpc outputs list:
Output 1 (Computer) is disabled
Output 2 (Creative T12 wireless) is enabled.

Then, after turning the speaker of and on again:
Output 1 (Computer) is disabled
Output 3 (Creative T12 wireless) is disabled.
Output 2 (Creative T12 wireless) is enabled.

After mpc enable 3, sound recovers...
Output 1 (Computer) is disabled
Output 3 (Creative T12 wireless) is enabled.
Output 2 (Creative T12 wireless) is enabled.

**Comment by ejurgensen** [comment]
Good to hear from you again, and thanks for the report.

I don't know what is going on here, so I have to ask for a log (debug level, please).

**Comment by LordMyschkin** [comment]
I am always here, enjoying progress of this great project.
Please give me a few days, it‘s my spare music- only laptop i dont have access to before the weekend…

**Comment by LordMyschkin** [comment]
Hi! Faster than expected: Could reproduce the issue with loglevel:debug:
[forked-daapd1.log.zip](https://github.com/owntone/owntone-server/files/6576981/forked-daapd1.log.zip)

**Comment by ejurgensen** [comment]
Looks like the index that Pulseaudio provides increases, which makes OwnTone think it is another speaker (the "id" in the end):
```
[2021-06-01 16:23:00] [DEBUG]   laudio: Callback for Pulseaudio sink 'bluez_sink.00_02_3C_26_6F_0C.a2dp_sink' (id 3)
[2021-06-01 16:24:25] [DEBUG]   laudio: Callback for Pulseaudio sink 'bluez_sink.00_02_3C_26_6F_0C.a2dp_sink' (id 4)
```
I suppose that is an indication that the index cannot be used as an id. I will have a look at Pulseaudio docs to see what would be a better way of identifying a speaker.

**Comment by ejurgensen** [comment]
I have now checked the docs, and I couldn't find an alternative, more persistent id to use. Well except using the name as an id, but that doesn't seem quite right (or?). One thing I would like you to test: When the speaker is just in the list once, run `pactl list sinks`, then turn if on/off, so it appears twice, and then again run `pactl list sinks`. Then please share both outputs here.

**Comment by LordMyschkin** [comment]
Here you are. PACTL does see it only once....
[pactl_lists.txt](https://github.com/owntone/owntone-server/files/6608202/pactl_lists.txt)

**Comment by ejurgensen** [comment]
Well, you can see that Pulseaudio also gives it a new ID:

```
Ziel #2
	Status: RUNNING
	Name: bluez_sink.00_02_3C_26_6F_0C.a2dp_sink
...
Ziel #3
	Status: RUNNING
	Name: bluez_sink.00_02_3C_26_6F_0C.a2dp_sink
```

So I think what I should be looking at is why owntone isn't removing the device when it is switched off. Perhaps a callback that is missing. I will look into it.

**Comment by ejurgensen** [comment]
I made an attempt at fixing this. Would you be able to build from the 'pulse_id1' branch and test the fix?

**Comment by LordMyschkin** [comment]
The behavoir changed, but it is not really consistent - but it's hard for me to tell if it OS BT-Implementation or Owntones Pulseaudio-implementation.
Right now, Owntone plays through BT although  mpc outputs shows it's disabled - seems it is mistaken for PCs main audio (which is shown as enabled although not playing).
Nice point though - after disconnection and reconnection, it is only shown once... I will try to do some more systematic tests and report then.

**Comment by ejurgensen** [comment]
Perhaps if you share a log I can see if there is anything that looks incorrect

**Comment by LordMyschkin** [comment]
Here is the actual log - would loglevel:debug help you?
[log_pulse.txt](https://github.com/owntone/owntone-server/files/6626871/log_pulse.txt)
btw, it did not get #1268 after you latest modification....

**Comment by ejurgensen** [comment]
Yes, it should be debug. And you need to catch the part where you start playing.

I'm pretty sure I haven't done anything to fix 1268 :-)

**Comment by LordMyschkin** [comment]
Well; here you are. 
As I said: BT-Speaker starts playing, Computerspeaker stops. 
After power-cycling the BT speakter, it gets captured, and start playing but sound stutters - maybe you can see why in the log. After disabling and re-coupling bt device M360, it plays smoothly.

All very strange; I hope this all will stop when I can install Elementary OS 6 in a few weeks...
[log_pulse2.txt](https://github.com/owntone/owntone-server/files/6626955/log_pulse2.txt)
